homeassistant/components/netatmo/__init__.py

```diff
@@ -83,15 +83,17 @@ def setup(hass, config):
     """Set up the Netatmo devices."""
     import pyatmo
 
-    global NETATMO_AUTH
     hass.data[DATA_PERSONS] = {}
     try:
-        NETATMO_AUTH = pyatmo.ClientAuth(
+        auth = pyatmo.ClientAuth(
             config[DOMAIN][CONF_API_KEY], config[DOMAIN][CONF_SECRET_KEY],
             config[DOMAIN][CONF_USERNAME], config[DOMAIN][CONF_PASSWORD],
             'read_station read_camera access_camera '
             'read_thermostat write_thermostat '
             'read_presence access_presence read_homecoach')
+
+        # Store config to be used during entry setup
+        hass.data[DATA_NETATMO_AUTH] = auth
```

Comment:
Move this out of the try... except, just after the except block. Sorry, I should have been more specific.